kickstarter/android-oss
# Messages in drawer

# What ❓
- Adding `Messages` to the navigation drawer in Discovery.
- When a user has an `unreadMessagesCount` > 0, we display the count to the right of the label.
- Since there are now 3 ways to get to the `MessageThreadsActivity`, we added `KoalaContext.Mailbox`. Users can access their mailbox from `creator_dashboard`, `drawer`, or `profile.
- `LoggedInViewHolder` didn't have a ViewModel so I added one so I could test!

# How to QA? 🤔
1. Open the app and verify the `Messages` row is only there for logged in users.
2. The unread count should display if it's greater than 0.
3. Clicking the row should take you to your inbox.

# Story 📖
[Trello Story](https://trello.com/c/toqzvUWn/1376-add-messages-to-android-drawer)

# See 👀
<img src="https://user-images.githubusercontent.com/1289295/58203818-0400f680-7ca9-11e9-8d84-7781ef062c77.png" width="330"/>
